DirectProgramming/C++SYCL_FPGA/Tutorials/GettingStarted/fpga_template/src/fpga_template.cpp

```diff
@@ -0,0 +1,114 @@
+#include <iostream>
+
+// oneAPI headers
+#include <sycl/ext/intel/fpga_extensions.hpp>
+#include <sycl/sycl.hpp>
+
+#include "exception_handler.hpp"
+
+// Forward declare the kernel name in the global scope. This is an FPGA best
+// practice that reduces name mangling in the optimization reports.
+class VectorAddID;
+
+class VectorAdd {
+ public:
+  int *a_in;
+  int *b_in;
+  int *c_out;
+  int len;
+
+  void operator()() const {
+    for (int idx = 0; idx < len; idx++) {
+      int a_val = a_in[idx];
+      int b_val = b_in[idx];
+      int sum = a_val + b_val;
+      c_out[idx] = sum;
```

Comment:
Why be more verbose here.    This is equivalent to 

c_out[idx] = a_in[idx] + b_in[idx]; 

This is not that long or complex so why do we need the extra temporaries the compiler will just need to optimize out.   Is this needed as part of an explanation of how the compiler works internally?

Reply:
correct.